stock_valuation_account_manual_adjustment/models/product.py

```diff
@@ -0,0 +1,230 @@
+# -*- coding: utf-8 -*-
+# Copyright 2016-18 Eficent Business and IT Consulting Services S.L.
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html).
+
+from odoo import api, exceptions, fields, models, _
+from odoo.addons import decimal_precision as dp
+import operator
+
+ops = {'=': operator.eq,
+       '!=': operator.ne,
+       '<=': operator.le,
+       '>=': operator.ge,
+       '>': operator.gt,
+       '<': operator.lt}
+
+UNIT = dp.get_precision('Account')
+
+
+class ProductProduct(models.Model):
+    _inherit = "product.product"
+
+    inventory_value = fields.Float(
+        string="Inventory Value", compute="_compute_inventory_account_value",
+        search="_search_inventory_value", digits=UNIT,
+        groups="stock_valuation_account_manual_adjustment."
+               "group_stock_valuation_account_manual_adjustment",
+    )
+    accounting_value = fields.Float(
+        string="Accounting Value", compute="_compute_inventory_account_value",
+        search="_search_accounting_value", digits=UNIT,
+        groups="stock_valuation_account_manual_adjustment."
+               "group_stock_valuation_account_manual_adjustment",
+    )
+    valuation_discrepancy = fields.Float(
+        string="Valuation discrepancy",
+        compute="_compute_inventory_account_value",
+        search="_search_valuation_discrepancy", digits=UNIT,
+        groups="stock_valuation_account_manual_adjustment."
+               "group_stock_valuation_account_manual_adjustment",
+    )
+
+    @api.multi
+    def _compute_inventory_account_value(self):
+        accounting_val = self._get_accounting_valuation_by_product()
+        inventory_val = self._get_inventory_valuation_by_product()
+        for product in self:
+            inventory_v = inventory_val.get(product.id, 0.0)
+            accounting_v = accounting_val.get(product.id, 0.0)
+            product.accounting_value = accounting_v
+            product.inventory_value = inventory_v
+            product.valuation_discrepancy = inventory_v - accounting_v
+
+    @api.model
+    def _get_accounting_valuation_by_product(self):
+        accounting_val = {}
+        self.env.cr.execute("""
```

Comment:
Honestly I don't have time to translate that query and give the time and love it would need... so, how about letting contributors know with a note in the roadmap?

Reply:
Ok, what I see now is that the query does not involves company_id on ir.property. So, I doubt this works in a multi company environment.

Reply:
Just added some notes to the roadmap!